xbmc/LangInfo.cpp

```diff
@@ -981,6 +981,16 @@ void CLangInfo::SetCurrentRegion(const std::string& strName)
 
   m_currentRegion->SetGlobalLocale();
 
+  // Check if locale is not affected by #19883
+  int test19883 = std::tolower('i') - std::tolower('I');
+  if (test19883 != 0)
```

Comment:
maybe change to
```c
const bool fallbackToDefaultRegionNeeded = (std::tolower('i') - std::tolower('I')) != 0;
if (fallbackToDefaultRegionNeeded)
```
plus a little comment/description why this check is needed would be awesome

Reply:
You can add `std::tolower('I')` to both sides and compare `(std::tolower('i') != std::tolower('I'));`